pkg/ccl/crosscluster/logical/logical_replication_job_test.go

```diff
@@ -1003,38 +1003,172 @@ func TestHeartbeatCancel(t *testing.T) {
 	require.ErrorContains(t, <-retryErrorChan, fmt.Sprintf("replication stream %s is not running, status is STREAM_INACTIVE", prodAID))
 }
 
-func setupLogicalTestServer(
-	t *testing.T, ctx context.Context, clusterArgs base.TestClusterArgs, numNodes int,
-) (
-	*testcluster.TestCluster,
-	serverutils.ApplicationLayerInterface,
-	*sqlutils.SQLRunner,
-	*sqlutils.SQLRunner,
-) {
+func TestAcyclicReplication(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	skip.UnderRace(t, "multi cluster/node config exhausts hardware")
+
+	ctx := context.Background()
+
+	clusterArgs := base.TestClusterArgs{
+		ServerArgs: base.TestServerArgs{
+			DefaultTestTenant: base.TestControlsTenantsExplicitly,
+			Knobs: base.TestingKnobs{
+				JobsTestingKnobs: jobs.NewTestingKnobsWithShortIntervals(),
+				DistSQL:          &execinfra.TestingKnobs{},
+			},
+		},
+	}
+
+	numDBs := 3
+	server, s, runners := setupServerWithNumDBs(t, ctx, clusterArgs, 1, numDBs)
+	defer server.Stopper().Stop(ctx)
+
+	PGURLs, cleanup := GetPGURLs(t, s, numDBs)
+	defer cleanup()
+
+	jobIDs := make([]jobspb.JobID, 3)
+	for i := range numDBs {
+		// Set a dummy jobID to make the array logic simpler
+		if i == 0 {
+			jobIDs[0] = jobspb.InvalidJobID
+			continue
+		}
+		runners[2].QueryRow(t, "CREATE LOGICAL REPLICATION STREAM FROM TABLE tab ON $1 INTO TABLE tab", PGURLs[i].String()).Scan(&jobIDs[i])
+	}
+
+	// Insert into C, then check B -> C wins
```

Comment:
instead of referencing the dbss in the docstring as "A","B","C", why not stick to their db names "a0","a1", and "a2". This will also make it easier to follow when indexing into `runners` and `jobIDs`

Reply:
When you say "stick to their db names", are they defined as a0, a1, a2 elsewhere? If not, I don't see how a0, a1, a2 is that much clearer than A,B,C

Reply:
First, the dbnames created are in fact `a0`, `a1`, `a2`, so i think documenting a different set of names is confusing.

Second, i find it easier to associate `a0` with `runner[0]` and `jobIDs[0]`, rather than `a` to index entry 0. In other words the numbered suffix in the dbname maps directly to its index in your helper arrays.

Reply:
ah yeah, i missed the rune thing. I still think it would be easier to follow explicitly name the db and jobIDs: e.g. `jobB`, `jobC`,  `sqlA`, `sqlB`, `sqlC`, rather than using the arrays you're using.

Reply:
updated